schema.go

```diff
@@ -535,33 +535,12 @@ func nodeOf(t reflect.Type, tag []string) Node {
 		}
 
 	case reflect.Map:
-		var mapTag, valueTag, keyTag string
-		if len(tag) > 0 {
-			mapTag = tag[0]
-			if len(tag) > 1 {
-				keyTag = tag[1]
-			}
-			if len(tag) >= 2 {
-				valueTag = tag[2]
-			}
+		if t.Implements(deprecated.MapGroupI) {
```

Comment:
How about moving the logic to the `deprecated` package so we don't have to export the sentinel interface used to determine the type?
```suggestion
		if deprecated.IsMap(t) {
```